samples/application_development/external_lib/README.rst

```diff
@@ -5,7 +5,34 @@ External Library
 
 Overview
 ********
+
 A simple example that demonstrates how to include an external static library
 into the Zephyr build system.
 The demonstrates both how to build the external library using a different build
 system and how to include the built static library.
+
+Windows Note
+************
+
+To use this sample on a Windows host operating system, GNU Make needs to be in
```

Comment:
Instead of documenting how to install Make on windows in a sample, could we give this sample a ninja build file to build it on windows?
Or maybe just call into a separate CMake build instead of a Makefile to make it portable.

Seeing as we recommend ninja on windows that seems like a better method.

Reply:
The purpose of the makefile is to demonstrate pulling in an external makefile-based project, i.e. mimicking many available libraries

Reply:
The purpose of the sample is to demonstrate including an external static library into the zephyr build system, using a different build system. It does not have to be Make.

I don't see the point in documenting how to get GNU Make on Windows to work as part of our documentation.

Reply:
It's setup for a single demo. It's no different than boards requiring usage of `srec` for some operations, e.g. https://docs.zephyrproject.org/latest/boards/arm/v2m_musca_s1/doc/index.html#uploading-an-application-to-v2m-musca-s1

Reply:
Not used it, I just remember it being a part of the TF-M requirements for nordic and other boards. This change request is related to customer issues with using this sample, if this sample should be changed to use ninja or something else can be done, but is outside the scope of this issue.

Reply:
I'm not blocking this update, just stating my opinion.

Mostly because I've had a lot of bad experience with Make on windows, I am sure this simple sample works, but if they are trying to use Makefile generated by cmake for zephyr it would most likely have problem.

I think going with converting the makefile to cmake would be a better solution.
Supporting a different build environment just for a sample is expensive in terms of documentation and support in my opinion.

I would also say that the srec documentation should be removed and instead documented using scripts/build/mergehex.py. 
For nordic platforms at least this documentation is not needed because the build system already performs this step. Maybe the musca-board also does this (or can be updated to do this) and this documentation can be removed.
See:
```
config TFM_FLASH_MERGED_BINARY
	bool
	help
	  This option instructs west flash to program the combined (merged)
	  binary consisting of the TF-M Secure firmware image, optionally, the
	  BL2 image (if building with TFM_BL2 is enabled), and the Non-Secure
	  application firmware.
```

Reply:
IMHOI It's more useful for the customer to have an example of how to integrate with Make than Ninja as Make is commonly used in embedded, but Ninja is not.

Reply:
this sample target users who *already* has a project based on something which is *not* CMake.

If those users are already running on Windows and building there library on that OS, then it means they already have a working setup.

If they already have a working setup to build their library, we don't need to inform them how to install `make`, cause we should expect them to already have a working environment.

So I don't see the value in telling them how to install this, cause if they don't know how to setup such environment, then basically they don't have a Makefile based project on Windows they need to integrate.

Reply:
For the record I am not blocking this.

My point is about whether we should maintain a guide on how to get make working on windows or not.
IMO this is just unnecessary documentation maintenance burden, and we should either not do it, or have an external project based on already documented tools, so that we can run this sample without additional dependencies on all platforms.

Reply:
> So I don't see the value in telling them how to install this, cause if they don't know how to setup such environment, then basically they don't have a Makefile based project on Windows they need to integrate.

There's been a lot of people running this that do not have a working makefile setup, apparently, hence the update here to fix it. A sample should explain how to use it, like include files, it should not require undocumented knowledge to run it.